modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -162,42 +504,797 @@ public void testListWithGetInt(String filter, int startIndexInt, int countInt,
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else {
+
+            abstractResourceManager.when(()
+                    -> AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+            doThrow(new NotFoundException()).when(userManager).deleteUser(id);
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        }
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserFailsINCharonExceptionOnly")
+    public Object[][] dataToTestDeleteUserCharonExceptionOnly()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        return new Object[][]{
+                {id, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteUserFailsINCharonExceptionOnly")
+    public void testDeleteUserCharonExceptionOnly(String id, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/" + USER_ID);
+
+        doThrow(new CharonException()).when(userManager).deleteUser(id);
+
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+
+        SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestUpdateWithPUTSuccess")
+    public Object[][] dataToTestUpdateWithPUTSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
 
         SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
         JSONDecoder decoder = new JSONDecoder();
-        return decoder.decodeResource(scimObjectString, schema, new User());
+
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+        String id = userOld.getId();
+
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_UPDATE, schema, new User());
+
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", null,
+                        userNew, userOld, ResponseCodeConstants.CODE_OK}
```

Comment:
consider datasets with different combinations for `attributes` and `excludeAttributes`

Reply:
Added in commit 6f5ca461fa07ac4b49182320325156a58b80b910